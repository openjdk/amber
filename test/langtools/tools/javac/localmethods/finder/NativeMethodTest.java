/**
 * @test /nodynamiccopyright/
 * @bug 8234792
 * @summary Finder should not recommend native methods as local method candidates
 * @compile/fail/ref=NativeMethodTest.out -Werror -XDrawDiagnostics -XDfind=local-methods NativeMethodTest.java
 */

public class NativeMethodTest {

    private static native void xxx();
    private static void yyy() {}
    private static void zzz() {}
    public static void main(String [] args) {
        xxx();
        yyy();
        zzz();
    }
    public static void main() {
        zzz();
    }
}
