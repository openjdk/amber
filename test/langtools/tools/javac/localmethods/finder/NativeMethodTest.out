NativeMethodTest.java:11:25: compiler.warn.method.could.be.declared.local: main(java.lang.String[]), kindname.class, NativeMethodTest
- compiler.err.warnings.and.werror
1 error
1 warning
